src/hotspot/share/opto/phaseX.cpp

```diff
@@ -1552,6 +1552,18 @@ void PhaseIterGVN::add_users_to_worklist( Node *n ) {
         }
       }
       if (use_op == Op_CmpI) {
+        // Put If on worklist to enable fold_compares optimization (see IfNode::cmpi_folds)
+        if (use->outcnt() > 0) {
+          for (uint i = 0; i < use->outcnt(); i++) {
```

Comment:
The `use->outcnt() > 0` check is redundant and you could use `DUIterator_Fast` instead of `raw_out`. Also, `bol` could have multiple if users. For an example, see code in `countedloop_phi_from_cmp`.